src/EditorFeatures/Core/Implementation/CodeFixes/ICodeFixService.cs

```diff
@@ -11,6 +11,7 @@ namespace Microsoft.CodeAnalysis.CodeFixes
     internal interface ICodeFixService
     {
         Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document document, TextSpan textSpan, bool includeSuppressionFixes, CancellationToken cancellationToken);
+        Task<CodeFixCollection> GetFixesAsync(Document document, TextSpan textSpan, string diagnosticId, CancellationToken cancellationToken);
```

Comment:
feels weird for the service to have both these methods.  Feels like we'd just want one method, and perhaps have these two as extension methods that both call into that one core method.  (note: i may be wrong about this).

Reply:
yes. we can merge this 2.

Reply:
actually, not sure. we can rename the method to GetFixAllForSpanAsync or something..

Reply:
I don't think we should add a new API to this service, given that it has very specific semantics for code cleanup. Can we instead just extend the existing `Task<ImmutableArray<CodeFixCollection>> GetFixesAsync` API to take a new flag `string diagnosticIdOpt`, and then have the code-cleanup handle passing in non-null diagnostic ID and performing `.FirstOrDefault()`? This suggestion can be implemented with a follow up PR as well.

---
In reply to: [192884230](https://github.com/dotnet/roslyn/pull/26802#discussion_r192884230) [](ancestors = 192884230)